### PR TITLE
Update the OpenID module to target the latest OpenIddict packages

### PR DESCRIPTION
**Pull request by PinpointTownes**
@jersiovic @sebastienros PTAL when you have a minute :smile:

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.OpenId/Services/OpenIdAuthorizationStore.cs [hunk 1]
```diff
@@ -1,36 +1,489 @@
...
+        /// <param name="count">The number of results to return.</param>
+        /// <param name="offset">The number of results to skip.</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> that can be used to abort the operation.</param>
+        /// <returns>
+        /// A <see cref="Task"/> that can be used to monitor the asynchronous operation,
+        /// whose result returns all the elements returned when executing the specified query.
+        /// </returns>
+        public virtual Task<ImmutableArray<OpenIdAuthorization>> ListInvalidAsync(int? count, int? offset, CancellationToken cancellationToken)
+        {
+            // TODO Sébastien Ros.
```
Comment (PinpointTownes) [inline src/OrchardCore.Modules/OrchardCore.OpenId/Services/OpenIdAuthorizationStore.cs]: @sebastienros as said on Gitter, I'll need your help with that :smile: 

For reference, here's the default implementation used by the EF/EF Core stores:

```csharp
/// <summary>
/// Lists the ad-hoc authorizations that are marked as invalid or have no
/// valid token attached and that can be safely removed from the database.
/// </summary>
/// <param name="count">The number of results to return.</param>
/// <param name="offset">The number of results to skip.</param>
/// <param name="cancellationToken">The <see cref="CancellationToken"/> that can be used to abort the operation.</param>
/// <returns>
/// A <see cref="Task"/> that can be used to monitor the asynchronous operation,
/// whose result returns all the elements returned when executing the specified query.
/// </returns>
public virtual Task<ImmutableArray<TAuthorization>> ListInvalidAsync([CanBeNull] int? count, [CanBeNull] int? offset, CancellationToken cancellationToken)
{
    IQueryable<TAuthorization> Query(IQueryable<TAuthorization> authorizations, int? skip, int? take)
    {
        var query = (from authorization in authorizations
                     where authorization.Status != OpenIddictConstants.Statuses.Valid ||
                          (authorization.Type == OpenIddictConstants.AuthorizationTypes.AdHoc &&
                          !authorization.Tokens.Any(token => token.Status == OpenIddictConstants.Statuses.Valid))
                     orderby authorization.Id
                     select authorization).AsQueryable();

        if (skip.HasValue)
        {
            query = query.Skip(skip.Value);
        }

        if (take.HasValue)
        {
            query = query.Take(take.Value);
        }

        return query;
    }

    return ListAsync((authorizations, state) => Query(authorizations, state.offset, state.count), (offset, count), cancellationToken);
}
```

File: src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AdminController.cs [hunk 2]
```diff
@@ -122,8 +121,8 @@ public class AdminController : Controller
             {
                 Id = id,
                 DisplayName = application.DisplayName,
-                RedirectUri = application.RedirectUri,
-                LogoutRedirectUri = application.LogoutRedirectUri,
+                RedirectUri = application.RedirectUris.FirstOrDefault(),
```
Comment (PinpointTownes) [inline src/OrchardCore.Modules/OrchardCore.OpenId/Controllers/AdminController.cs]: FYI I opened a follow-up ticket to track possible future improvements: https://github.com/OrchardCMS/OrchardCore/issues/1287.

**Comment by jersiovic** [comment]
Great job, thank you @PinpointTownes

File: src/OrchardCore.Modules/OrchardCore.OpenId/Services/OpenIdAuthorizationStore.cs [hunk 3]
```diff
@@ -348,10 +349,23 @@ public virtual Task<OpenIdAuthorization> InstantiateAsync(CancellationToken canc
         /// A <see cref="Task"/> that can be used to monitor the asynchronous operation,
         /// whose result returns all the elements returned when executing the specified query.
         /// </returns>
-        public virtual Task<ImmutableArray<OpenIdAuthorization>> ListInvalidAsync(int? count, int? offset, CancellationToken cancellationToken)
+        public virtual async Task<ImmutableArray<OpenIdAuthorization>> ListInvalidAsync(int? count, int? offset, CancellationToken cancellationToken)
         {
-            // TODO Sébastien Ros.
-            throw new NotImplementedException();
+            IQuery< OpenIdAuthorization> query = _session.Query<OpenIdAuthorization, OpenIdAuthorizationIndex>(authorization => authorization.Status != OpenIddictConstants.Statuses.Valid ||
```
Comment (PinpointTownes) [inline src/OrchardCore.Modules/OrchardCore.OpenId/Services/OpenIdAuthorizationStore.cs]: @sebastienros I'm glad you implemented that yourself :sweat_smile: 

Merci.

Comment (sebastienros) [reply src/OrchardCore.Modules/OrchardCore.OpenId/Services/OpenIdAuthorizationStore.cs]: I haven't tried it. I am also changing the identifiers to logical ones right now.

**Comment by sebastienros** [comment]
@PinpointTownes did it work for you?

**Comment by PinpointTownes** [comment]
@sebastienros nope. I posted more details on Gitter :smile:

**Comment by PinpointTownes** [comment]
I went ahead and merged this PR. I'll submit a separate PR for the changes needed to decouple the OpenID module from YesSql and make writing custom stores modules easier.

**Comment by Jetski5822** [comment]
nice :)

**Pull request merged**